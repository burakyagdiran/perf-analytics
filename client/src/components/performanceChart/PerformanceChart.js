import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import Chart from 'react-apexcharts';
import moment from 'moment';

function PerformanceChart({
  title,
  performanceData,
  performanceType,
  ...props
}) {
  const [options, setOptions] = useState({});
  const [series, setSeries] = useState([]);

  useEffect(() => {
    let newOptions = {
      title: { text: title },
      chart: {
        toolbar: {
          show: true,
          offsetX: 0,
          offsetY: 0,
          tools: { download: false, zoomin: false, zoomout: false, pan: false },
          autoSelected: 'zoom',
        },
      },
      xaxis: { categories: [], title: { text: 'Occurrence Time' } },
      yaxis: {
        title: { text: 'Time in millisecond' },
        labels: {
          formatter: function (x) {
            return x;
          },
        },
      },
    };
    let newSeries = [
      {
        data: [],
      },
    ];
    performanceData.forEach((item) => {
      newOptions.xaxis.categories.push(
        moment(item.createDate).format('HH:mm:ss')
      );
      newSeries[0].data.push(item[performanceType]);
    });
    setOptions(newOptions);
    setSeries(newSeries);
    // eslint-disable-next-line
  }, [title, performanceData, performanceType]);
  return (
    <Chart
      options={options}
      series={series}
      type="line"
      width="100%"
      height="100%"
    />
  );
}

PerformanceChart.propTypes = {
  title: PropTypes.string,
  performanceType: PropTypes.string,
  performanceData: PropTypes.array.isRequired,
};

PerformanceChart.defaultProps = {
  performanceData: [],
};

export default PerformanceChart;
