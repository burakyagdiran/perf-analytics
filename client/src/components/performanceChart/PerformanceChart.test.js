import React from 'react';
import { shallow } from 'enzyme';
import PerformanceChart from './PerformanceChart';

it('should render correctly with no props', () => {
  const component = shallow(<PerformanceChart />);
  expect(component).toMatchSnapshot();
});

it('should render banner text correctly with given strings', () => {
  const testTitle = 'test title';
  const component = shallow(<PerformanceChart title={testTitle} />);
  expect(component).toMatchSnapshot();
});

describe('PerformanceChart', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<PerformanceChart debug />);
    expect(component).toMatchSnapshot();
  });
});
