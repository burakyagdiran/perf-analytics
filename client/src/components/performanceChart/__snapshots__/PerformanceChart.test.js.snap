// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PerformanceChart should render correctly in "debug" mode 1`] = `
<r
  height="100%"
  options={Object {}}
  series={Array []}
  type="line"
  width="100%"
/>
`;

exports[`should render banner text correctly with given strings 1`] = `
<r
  height="100%"
  options={Object {}}
  series={Array []}
  type="line"
  width="100%"
/>
`;

exports[`should render correctly with no props 1`] = `
<r
  height="100%"
  options={Object {}}
  series={Array []}
  type="line"
  width="100%"
/>
`;
